daphne_worker/src/storage_proxy.rs

```diff
@@ -71,23 +71,85 @@
 //!
 //! [to_uri]: daphne_service_utils::durable_requests::bindings::DurableMethod::to_uri
 
-use std::time::Duration;
+use std::{sync::OnceLock, time::Duration};
 
-use daphne_service_utils::durable_requests::{
-    DurableRequest, ObjectIdFrom, DO_PATH_PREFIX, KV_PATH_PREFIX,
+use daphne::auth::BearerToken;
+use daphne_service_utils::{
+    auth::TlsCertInfo,
+    durable_requests::{DurableRequest, ObjectIdFrom, DO_PATH_PREFIX, KV_PATH_PREFIX},
 };
 use tracing::warn;
 use url::Url;
 use worker::{js_sys::Uint8Array, Delay, Env, Request, RequestInit, Response};
 
 const KV_BINDING_DAP_CONFIG: &str = "DAP_CONFIG";
 
+pub fn unauthorized_reason(req: &Request, env: &Env) -> Option<worker::Result<Response>> {
```

Comment:
Make sure this matches the dap_prototype. In particular:
* we're missing the check for the special "SUCCESS" string
* Pass a `String` to  "access_denied()"  that describes the failure reason

Reply:
We also decided in our sync to:
* rip out the TLS client auth bits
* change the header to be more standard